tests/conftest.py

```diff
@@ -271,6 +271,16 @@ def __init__(
                 ).to(dtype=torch_dtype))
         else:
             model_kwargs = model_kwargs if model_kwargs is not None else {}
+            config = AutoConfig.from_pretrained(
+                model_name,
+                torch_dtype=torch_dtype,
+                trust_remote_code=True,
+            )
+            arch = config.architectures 
+            if len(arch) > 0:
+                cls_type = arch[0].split("For")[-1]
+                auto_cls = eval(f"AutoModelFor{cls_type}")
+            
```

Comment:
This looks unsafe. We can already specify the class to use using `auto_cls` argument, so there is no need to do this.

Reply:
Thank you for your comments.
I want to take your opinion related to this issue.

the `auto_cls` is registered as "AutoModelForCausalLM" by default.
However, in this scene, we want to register as "AutoModelForSequenceClassification".
What do you think about the best option for this situation & context?

Reply:
You can manually pass `auto_cls=AutoModelForSequenceClassification` when calling `hf_runner`.

Reply:
For example, the BART tests use the code

```
    with hf_runner(model, dtype=dtype,
                   auto_cls=AutoModelForSeq2SeqLM) as hf_model:
```

In your tests, you can use `AutoModelForSequenceClassification` instead.

Reply:
Thank you for your suggestion.
This logic has been commited.